starter-code/basic-algorithms.js

```diff
@@ -8,29 +8,84 @@ console.log(`The navigator's name is ${hacker2}`);
 //Conditionals
 
 if (hacker1.length > hacker2.length)
-  console.log(`The Driver has the longest 
-name it has ${hacker1.length} characters `);
+  console.log(
+    `The Driver has the longest name it has ${hacker1.length} characters `
+  );
 if (hacker1.length < hacker2.length)
-  console.log(`The Navigator has the longest
-name, it has ${hacker2.length} characters `);
+  console.log(
+    `The Navigator has the longest name, it has ${hacker2.length} characters `
+  );
 if (hacker1.length === hacker2.length)
-  console.log(`wow, you got equally long names
-, ${hacker1.length} characters`);
-
-// let str = "This is nothing much is yes is is is is ";
-// let tab = str.split(" ");
-
-// function findRemove(str, tab) {
-//   let clone = [...tab];
-//   let index = tab.indexOf(str);
-//   let count = 0;
-//   while (index !== -1) {
-//     count++;
-//     clone.splice(index, 1);
-//     index = clone.indexOf(str);
-//   }
-//   return count;
-// }
-
-// console.log(findRemove("is", tab));
+  console.log(
+    `wow, you got equally long names, ${hacker1.length} characters !!`
+  );
+
+//Loops
+
+let result = [];
+for (let i = 0; i < hacker1.length; i++) {
+  result.push(hacker1.charAt(i).toUpperCase(hacker1.charAt(i)));
+}
+console.log(result.join(" "));
+//ehh
+console.log(
+  hacker2
+    .split("")
+    .reverse()
+    .join("")
+);
+
+if (hacker1.toLowerCase() < hacker2.toLowerCase())
+  console.log(`The driver's name goes first`);
+else if (hacker1.toLowerCase() > hacker2.toLowerCase())
+  console.log(`Yo, the navigator goes first definitely`);
+else if (hacker1.toLowerCase() == hacker2.toLowerCase())
+  console.log(`What?! You both got the same name?`);
+
```

Comment:
This is good but the problem is that if you put everything to lowercase, you change the lexicographic value of the string. A Capital letter doesn't have the same value as a lower case letter in the lexicographic environment.